linode/helper/sdkv2_cty.go

```diff
@@ -0,0 +1,72 @@
+package helper
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// SDKv2UnwrapOptionalConfigAttr returns a pointer to a value from the given path
+// only if the value was explicitly defined.
+//
+// If the path cannot be resolved, the function will return nil as
+// this may be valid case for nested fields.
+//
+// This is useful to simplifying the logic around checking whether a nested field has been
+// explicitly defined by a user.
+func SDKv2UnwrapOptionalConfigAttr[T any](ctx context.Context, d *schema.ResourceData, path string) *T {
+	ctyPath := SDKv2PathToCtyPath(path)
+
+	ctyValue, err := ctyPath.Apply(d.GetRawConfig())
+	if err != nil {
+		// This is a valid case, but we should log it just in case something was expected to be
+		// resolved
+		tflog.Trace(ctx, "failed to resolve path, assuming undefined", map[string]any{
+			"path": path,
+		})
+		return nil
+	}
+
+	// If the value hasn't been defined by the user, return nil
+	if !ctyValue.IsKnown() || ctyValue.IsNull() {
+		return nil
+	}
+
+	// If the value was defined explicitly as nil, return nil
+	value := d.Get(path)
+	if value == nil {
+		return nil
+	}
+
+	castedValue := value.(T)
+	return &castedValue
+}
```

Comment:
We can apply this to any user-defined SDKv2 bool field that we want to differentiate between false and undefined values on (e.g. booted). To keep the scope of this PR simple, I only made use of it for compliant_only 👍